libraries/AP_RTC/AP_RTC.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <stdint.h>
+
+class AP_RTC {
+
+public:
+
+    AP_RTC() {
+        if (_singleton != nullptr) {
+            // it's an error to get here.  But I don't want to include
+            // AP_HAL here
+            return;
+        }
+        _singleton = this;
+    }
+
+    typedef uint8_t source_id_t;
+    typedef enum : uint8_t {
+        SOURCE_GPS,
+        SOURCE_MAVLINK_SYSTEM_TIME,
+        SOURCE_HW,
+    } source_type;
+
+    /*
+      set clock in UTC microseconds
+     */
+    void set(source_id_t source_id, uint64_t time_utc_usec);
+
+    /*
+      check if clock was already set
+     */
+    bool is_set() const;
+
+    /*
+      get clock in UTC microseconds
+
+      non-const as it updates the RTC shift as part of retrieving the value
+     */
+    uint64_t get();
```

Comment:
I would probably rename this method to "get_system_time_usec" or something like that but it's not critical.

Reply:
I'd like to see it renamed too. I'm also dubious about it being non-const and triggering updates. Peter said its because I suggested it, but I don't remember that